.github/workflows/spelling2.yml

```diff
@@ -12,21 +12,83 @@ on:
     tags-ignore:
       - "**"
     types: ['opened', 'reopened', 'synchronize']
+  issue_comment:
+    types: [created]
 
 jobs:
   spelling:
     name: Spell checking
+    permissions:
+      contents: read
+      pull-requests: read
+      actions: read
+    outputs:
+      followup: ${{ steps.spelling.outputs.followup }}
     runs-on: ubuntu-latest
+    if: "contains(github.event_name, 'pull_request') || github.event_name == 'push'"
+    concurrency:
+      group: spelling-${{ github.event.pull_request.number || github.ref }}
+      # note: If you use only_check_changed_files, you do not want cancel-in-progress
+      cancel-in-progress: true
     steps:
-    - name: checkout-merge
-      if: "contains(github.event_name, 'pull_request')"
-      uses: actions/checkout@v2
-      with:
-        ref: refs/pull/${{github.event.pull_request.number}}/merge
-    - name: checkout
-      if: ${{ github.event_name == 'push' }}
-      uses: actions/checkout@v2
-    - uses: check-spelling/check-spelling@v0.0.19
+    - name: check-spelling
       id: spelling
+      uses: check-spelling/check-spelling@v0.0.20
+      with:
+        config: .github/actions/spell-check
+        experimental_apply_changes_via_bot: ${{ github.repository_owner != 'powerdns' && 1 || 0 }}
+        suppress_push_for_open_pull_request: 1
+        checkout: true
+        post_comment: 0
+        extra_dictionaries:
+          cspell:aws/aws.txt
+          cspell:filetypes/filetypes.txt
+          cspell:django/django.txt
+          cspell:html/html.txt
+          cspell:fullstack/fullstack.txt
+          cspell:npm/npm.txt
+          cspell:lua/lua.txt
+          cspell:python/python.txt
+          cspell:css/css.txt
+        check_extra_dictionaries: ''
+
+  comment:
+    name: Report
+    runs-on: ubuntu-latest
+    needs: spelling
+    permissions:
+      contents: write
+      pull-requests: write
+    if: (success() || failure()) && needs.spelling.outputs.followup
+    steps:
+    - name: comment
+      uses: check-spelling/check-spelling@v0.0.20
+      with:
+        config: .github/actions/spell-check
+        checkout: true
+        task: ${{ needs.spelling.outputs.followup }}
+        experimental_apply_changes_via_bot: ${{ github.repository_owner != 'powerdns' && 1 || 0 }}
+
+  update:
+    name: Update PR
+    permissions:
+      contents: write
+      pull-requests: write
```

Comment:
This could create commits on branches (as well as interacting w/ comments on PRs or making comments when things go wrong).